app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  belongs_to :user
+
+  def post_counter
+    user.update(post_counter: user.posts.count)
+  end
```

Comment:
- [ ] Kindly I recommend you to  use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the **#post_counter** method every time a new post has been saved to database, then make the **post_counter** method private Just like below.

```
  def recent_comments
    comments.order(created_at: :desc).limit(5)
  end

after_save :post_counter

  private

  def post_counter
        user.update(post_counter: user.posts.count)
  end

```